chore(README): more API oriented (#15)
JsCommunity/reaclette: README.md

      <button onClick={effects.addOne}>Add one</button>
    </p>
  </div>
));

render(<Parent />, document.body);
```

## API

```js
import { provideState, injectState } from "reaclette";
```

### `provideState(options) => (Component => Component)`

> Create a decorator which associate a React component with a store.

Reviewer: ```suggestion
Create a decorator that associates a React component with a store.
```

> `options` is an object which can contains the following properties.

Reviewer: ```suggestion
`options` is an object which can contain the following properties.
```

#### `initialState(props) => object`

This function returns the initial state of store, which can be computed from the properties of the decorated component.

```js
{
  initialState: (props) => ({ counter: 0 }),
}
```

#### `effects: { [string]: Effect }`

> These functions which can be called from application code (see `injectState`) and can do side-effects and/or mutate the state.

Reviewer: ```suggestion
These functions can be called from application code (see `injectState`) and can do side-effects and/or mutate the state.
```

When called, an effect is provided with one or more arguments: a reference to other effects and any arguments passed to the effect from the application code.

An effect can return:

1. nothing (`undefined` or `null`)
2. an object containing new properties to merge in the state
3. a promise resolving to one of the above

> 4. a function which receive the state and the props and returns one of the above

Reviewer: ```suggestion
4. a function which receives the state and the props then returns one of the above
```

## Contributions

> Contributions are _very_ welcomed, either on the documentation or on

Reviewer: ```suggestion
Contributions are _very_ welcomed, either to the documentation or to
```
